### PR TITLE
UHW fetch documents Gateway

### PR DESCRIPTION
**Pull request by kosiakkatrina**
Add UHW Fetch documents gateway and tests. Tested locally.

### CODE REVIEW

File: api/index.js [hunk 1]
```diff
@@ -136,6 +136,16 @@ app.get('/customers/:id/documents', async (req, res) => {
       getSystemId
     }
   );
+
+  const UHWFetchDocumentsGateway = require('./lib/gateways/UHW/UHWFetchDocuments')(
```
Comment (mmmmillar) [inline api/index.js]: lets keep all our casing the same (uhwFetch...)

File: api/lib/gateways/UHW/UHWFetchDocuments.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      { id: 'id', type: 'Int', value: id }
+    ]);
+  };
+
+  const processDocuments = results => {
+    return results.map(doc => {
+      return buildDocument({
+        id: doc.DocNo,
+        title: 'Document',
+        text: doc.DocDesc + `${doc.title ? ' - ' + doc.title : ''}`,
```
Comment (mmmmillar) [inline api/lib/gateways/UHW/UHWFetchDocuments.js]: mixing string concatenation and interpolation (let go all-out interpolation)

File: api/test/gateways/UHW/UHWFetchDocuments.test.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    expect(documents.length).toBe(1);
+  });
+
+  it('returns an empty set of records if there is an error', async () => {
+    const document = { DocNo: 1231 };
+    const gateway = createGateway([document], true);
+
+    const documents = await gateway.execute('1');
+
+    expect(documents.length).toBe(0);
```
Comment (mmmmillar) [inline api/test/gateways/UHW/UHWFetchDocuments.test.js]: should probably check that buildDocument hasnt been called

File: api/test/gateways/UHW/UHWFetchDocuments.test.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  });
+
+  it('builds a document', async () => {
+    const document = { DocNo: '1231' };
+    const gateway = createGateway([document]);
+
+    const documents = await gateway.execute({});
+    const paramMatcher = expect.objectContaining({ id: '1231' });
+    expect(buildDocument).toHaveBeenCalledWith(paramMatcher);
+    expect(documents.length).toBe(1);
```
Comment (mmmmillar) [inline api/test/gateways/UHW/UHWFetchDocuments.test.js]: Is this line necessary?

File: api/test/gateways/UHW/UHWFetchDocuments.test.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
+const UHWFetchDocuments = require('../../../lib/gateways/UHW/UHWFetchDocuments');
+
+describe('UHWFetchDocumentsGateway', () => {
+  let buildDocument;
+  let db;
+
+  const createGateway = (records, throwsError) => {
+    buildDocument = jest.fn(({}) => {
```
Comment (mmmmillar) [inline api/test/gateways/UHW/UHWFetchDocuments.test.js]: pretty sure we can simplify this to buildDocument = jest.fn(). what do you think?

File: api/index.js (hunk 1 above)
Comment (kosiakkatrina) [reply api/index.js]: I'll fix that in next commit, after shifting it to libDependencies, cause some of the consts in libdependencies are uppercase too. Is that cool?

File: api/test/gateways/UHW/UHWFetchDocuments.test.js (hunk 4 above)
Comment (kosiakkatrina) [reply api/test/gateways/UHW/UHWFetchDocuments.test.js]: Not sure which line you're talking about but if it's 55, the I think so? It checks that it only creates 1 document and not more. What are the downsides of having it?

File: api/lib/gateways/UHW/UHWFetchDocuments.js (hunk 2 above)
Comment (kosiakkatrina) [reply api/lib/gateways/UHW/UHWFetchDocuments.js]: I changed it bu I will revert it now. It seems that there was a reason for it (the conditional in the middle (?)): 

image

Comment (kosiakkatrina) [reply api/lib/gateways/UHW/UHWFetchDocuments.js]: Unless I misunderstood something

Comment (mmmmillar) [reply api/lib/gateways/UHW/UHWFetchDocuments.js]: I meant it should be `${doc.DocDesc}${doc.title ? ' - ' + doc.title : ''}`

File: api/test/gateways/UHW/UHWFetchDocuments.test.js (hunk 4 above)
Comment (mmmmillar) [reply api/test/gateways/UHW/UHWFetchDocuments.test.js]: Line 54 checks that it is called once, line 55 kinda tests that our fake implementation returns something

**Pull request merged**